### PR TITLE
[NEXUS-4530] test repo keyword search failure

### PR DESCRIPTION
**Pull request by adreghiciu**

### CODE REVIEW

File: nexus/nexus-app/src/test/java/org/sonatype/nexus/AbstractMavenRepoContentTests.java [hunk 1]
```diff
@@ -140,6 +140,11 @@ public abstract class AbstractMavenRepoContentTests
                 super.getContainer().getContext().get( PROXY_SERVER_PORT )+"/apache-snapshots/" );
         ( (MavenProxyRepository) apacheSnapshots ).setDownloadRemoteIndexes( false );
         nexusConfiguration.saveConfiguration();
+        // saving the configuration will trigger an re-indexing task for apache-snapshots
+        // wait for this tasks to run
+        // waiting was introduced as a fix for NEXUS-4530 but will benefit all subclasses of this test as it will
+        // avoid unexpected behavior
+        waitForTasksToStop();
```
Comment (velo) [inline nexus/nexus-app/src/test/java/org/sonatype/nexus/AbstractMavenRepoContentTests.java:148]: Just checking, but shouldn't you events.waitForQuietPeriod() as well?

Comment (adreghiciu) [inline nexus/nexus-app/src/test/java/org/sonatype/nexus/AbstractMavenRepoContentTests.java:148]: That part is not involved in issue in cause but that will not hurt. I will add it.

Comment (adreghiciu) [inline nexus/nexus-app/src/test/java/org/sonatype/nexus/AbstractMavenRepoContentTests.java:148]: Implemented in 4260e9a2

**Comment by velo** [comment]
+1

File: nexus/nexus-app/src/test/java/org/sonatype/nexus/AbstractMavenRepoContentTests.java [hunk 2]
```diff
@@ -140,6 +140,12 @@ public abstract class AbstractMavenRepoContentTests
                 super.getContainer().getContext().get( PROXY_SERVER_PORT )+"/apache-snapshots/" );
         ( (MavenProxyRepository) apacheSnapshots ).setDownloadRemoteIndexes( false );
         nexusConfiguration.saveConfiguration();
+        // saving the configuration will trigger an re-indexing task for apache-snapshots
+        // wait for this tasks to run
+        // waiting was introduced as a fix for NEXUS-4530 but will benefit all subclasses of this test as it will
+        // avoid unexpected behavior
+        wairForAsyncEventsToCalmDown();
+        waitForTasksToStop();
     }
```
Comment (peterlynch) [inline nexus/nexus-app/src/test/java/org/sonatype/nexus/AbstractMavenRepoContentTests.java:149]: We should get in the habit of adding missing javadoc as we touch old files, especially on public methods such as fillInRepo(). Would you mind adding some that summarize what is going on in this method? method name fillInRepo tells me nothing....

**Comment by nabcos** [comment]
+1

**Comment by cstamas** [comment]
+1

**Pull request merged**